app/models/concerns/user_encrypted_email_overrides.rb

```diff
@@ -0,0 +1,59 @@
+module UserEncryptedEmailOverrides
+  extend ActiveSupport::Concern
+
+  # override some Devise methods to support our use of encrypted_email
+
+  class_methods do
+    def find_first_by_auth_conditions(tainted_conditions, opts = {})
+      if tainted_conditions[:email].present?
+        opts[:email_fingerprint] = create_fingerprint(tainted_conditions.delete(:email))
+      end
+      to_adapter.find_first(devise_parameter_filter.filter(tainted_conditions).merge(opts))
+    end
+
+    def find_with_email(email)
+      email_fingerprint = create_fingerprint(email)
+      find_by(email_fingerprint: email_fingerprint)
+    end
+
+    def create_fingerprint(email)
```

Comment:
I get a reek warning here. Inclined to override and exclude it, because this is a class method that reek misidentifies.